lib/scholar/neighbors/radius_nearest_neighbors.ex

```diff
@@ -0,0 +1,347 @@
+  Makes predictions with the given `model` on inputs `x`.
+
+  ## Return Values
+
+    It returns a tensor with predicted class labels
+
+  ## Examples
+
+      iex> x = Nx.tensor([[1, 2], [2, 4], [1, 3], [2, 5]])
+      iex> y = Nx.tensor([1, 0, 1, 1])
+      iex> model = Scholar.Neighbors.RadiusNearestNeighbors.fit(x, y, num_classes: 2)
+      iex> Scholar.Neighbors.RadiusNearestNeighbors.predict(model, Nx.tensor([[1.9, 4.3], [1.1, 2.0]]))
+      Nx.tensor(
+        [0, 1]
+      )
+  """
+  defn predict(%__MODULE__{labels: labels, weights: weights, task: task} = model, x) do
+    case task do
+      :classification ->
+        {probabilities, outliers_mask} = predict_proba(model, x)
+        results = Nx.argmax(probabilities, axis: 1)
+        Nx.select(outliers_mask, -1, results)
+
+      :regression ->
+        {distances, indices} = radius_neighbors(model, x)
+
+        x_num_samples = Nx.axis_size(x, 0)
+        train_num_samples = Nx.axis_size(labels, 0)
+        labels_rank = Nx.rank(labels)
+
+        labels =
+          if labels_rank == 1 do
+            Nx.new_axis(labels, 0) |> Nx.broadcast({x_num_samples, train_num_samples})
+          else
+            out_size = Nx.axis_size(labels, 1)
+            Nx.new_axis(labels, 0) |> Nx.broadcast({x_num_samples, train_num_samples, out_size})
+          end
+
+        indices =
+          if labels_rank == 2,
+            do: Nx.new_axis(indices, -1) |> Nx.broadcast(labels),
+            else: indices
+
+        case weights do
+          :distance ->
+            weights = check_weights(distances)
+
+            weights =
+              if labels_rank == 2,
+                do: Nx.new_axis(weights, -1) |> Nx.broadcast(labels),
+                else: weights
+
+            Nx.weighted_mean(labels, indices * weights, axes: [1])
+
+          :uniform ->
+            Nx.weighted_mean(labels, indices, axes: [1])
+        end
+    end
+  end
+
+  @doc """
+  Return probability estimates for the test data `x`.
+
+  ## Return Values
+
+    It returns a typle with tensor with probabilities of classes and mask of outliers.
+    They are arranged in lexicographic order.
+
+
+  ## Examples
+
+      iex> x = Nx.tensor([[1, 2], [2, 4], [1, 3], [2, 5]])
+      iex> y = Nx.tensor([1, 0, 1, 1])
+      iex> model = Scholar.Neighbors.RadiusNearestNeighbors.fit(x, y, num_classes: 2)
+      iex> Scholar.Neighbors.RadiusNearestNeighbors.predict_proba(model, Nx.tensor([[1.9, 4.3], [1.1, 2.0]]))
+      {Nx.tensor(
+        [
+          [0.5, 0.5],
+          [0.0, 1.0]
+        ]
+      ),
+      Nx.tensor(
+        [0, 0], type: :u8
+      )}
+  """
+  deftransform predict_proba(
+                 %__MODULE__{
+                   task: :classification
+                 } = model,
+                 x
+               ) do
+    predict_proba_n(model, x)
+  end
+
+  @doc """
+  Find the Radius neighbors of a point.
+
+  ## Return Values
+
+    Returns indices of and distances to the neighbors of each point.
```

Comment:
```suggestion
    Returns indices of the selected neighbor points and their respective distances.
```

Reply:
```suggestion
    Returns indices of the selected neighbor points as a mask  (1 if a point is a neighbor, 0 otherwise) and their respective distances.
```